### PR TITLE
fix: Concurrency issues with Global Registry, issues with incorrect Path in Metrics

### LINKED ISSUE #187: incorrect path when using groups
Hi there,
thanks for this work!

When using `a.Group` the path is not correct.

With the following app
```go
app := fiber.New()

prometheus := fiberprometheus.NewWithLabels(labels, strings.ReplaceAll(Component, "-", "_"), "http")
prometheus.RegisterAt(app, "/metrics")
app.Use(prometheus.Middleware)

public := app.Group("/public")

public.Get("/somepath", func(c *fiber.Ctx) error {
   ...
})

```

The following metrics are created when calling `GET /public/somepath`

```lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="1e-09"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="2e-09"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="5e-09"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="1e-08"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="2e-08"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="5e-08"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="1e-07"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="2e-07"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="5e-07"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="1e-06"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="2e-06"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="5e-06"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="1e-05"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="2e-05"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="5e-05"} 0
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.0001"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.0002"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.0005"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.001"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.002"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.005"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.01"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.02"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.05"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.1"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.2"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="0.5"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="1"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="2"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="5"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="10"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="15"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="20"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="30"} 1
lessons_cdn_http_request_duration_seconds_bucket{build="development",method="GET",path="/public",status_code="401",version="development",le="+Inf"} 1
lessons_cdn_http_request_duration_seconds_sum{build="development",method="GET",path="/public",status_code="401",version="development"} 8.253e-05
lessons_cdn_http_request_duration_seconds_count{build="development",method="GET",path="/public",status_code="401",version="development"} 1
```

I'd expect to see `/public/somepath` in the `path` label.

Thanks!

### PR DESCRIPTION
**Pull request by gaby**
- Fixed concurrency issues related to using the Prometheus Global Registry. Each instance now uses it's own registry
- Fixed issue with Path in metrics, it now uses the Path from the Request `ctx.Request().RequestURI()`.
- Added benchmarks
- Added unit-tests for group handlers
- Replaced deprecated adaptor middleware with the one in https://github.com/gofiber/fiber

Fixes #187 #191 
Closes #195 #196 

@ansrivas I figured fixing these issues here first before migrating the middleware would make more sense. That way anyone using this repo will still get them.

### CODE REVIEW

File: middleware.go
Comment (sixcolors) [inline middleware.go:1]: rename `middleware.go` to `fiberprometheus.go`

Comment (gaby) [reply middleware.go:1]: It will be renamed when moved to gofiber/contrib.

**Review by ansrivas — APPROVED** [review]

**Comment by sashayakovtseva** [comment]
Seems like this PR broke existing code and http metrics will not be collected for users who used default registry and not used RegisterAt (for instance, one could have a separate metrics server, created elsewhere apart from fiberprometheus.New)

**Comment by gaby** [comment]
@sashayakovtseva That was a bug, the middleware was sharing a global variable. It was exposed when running test in parallel.

You can pass your current registry to the middleware and they both will use the same using `NewWithRegistry()`

**Comment by gaby** [comment]
Another option would be adding a config boolean to allow `New` use default registry or not. We are moving this middleware to https://github.com/gofiber/contrib I will implement that change there

**Comment by sashayakovtseva** [comment]
> You can pass your current registry to the middleware and they both will use the same using `NewWithRegistry()`

Yeah, I found that out already when metrics disappeared :) 

> the middleware was sharing a global variable

I guess that is an issue if one has multiple `fiberprometheus.New` calls concurrently, correct?

**Comment by gaby** [comment]
> > You can pass your current registry to the middleware and they both will use the same using `NewWithRegistry()`
> 
> Yeah, I found that out already when metrics disappeared :)
> 
> > the middleware was sharing a global variable
> 
> I guess that is an issue if one has multiple `fiberprometheus.New` calls concurrently, correct?

Yes, if you have multiple prometheus-go clients they all use the same `defaultRegistry`. So one solution is to create a Registry a pass it to all your clients.

**Pull request merged**